### PR TITLE
 🐛 [#1353] Allow nested datumkenmerk for afleidingswijze zaakobject 

### LINKED ISSUE #1353: Sluiten van een zaak met brondatum afleidingswijze zaakobject niet mogelijk in combinatie met URLs naar de ObjectsAPI of in combinatie met datum_kenmerk en lokale 'overige' JSON opslag
**General information**
Open Zaak version: 1.8.2

**Describe the bug**
Stel we hebben een zaak met een zaaktype waarbij het gewenste resultaattype op brondatum-afleidingswijze 'zaakobject' staat. De werking is dat er dan via het zaakobject wordt gecontroleerd of het ingestelde datum_kenmerk opvraagbaar is:

https://github.com/open-zaak/open-zaak/blob/ae1d11401a4f615cb5b9619b3273629c2256c03e/src/openzaak/components/zaken/brondatum.py#L129

Er zijn 2 mogelijkheden:

1. De externe object URL is gedefinieerd in `zaak_object.object`, deze wordt opgevraagd. Echter, datum_kenmerk kan alleen een attribuut zijn in de root van de opgehaalde JSON: {"einddatum": X}. In de Objecten API zit deze informatie per definitie dieper in het JSON structuur: {"record": {"einddatum": X}}. Deze diepere structuur wordt echter door Open Zaak genegeerd en het datum_kenmerk moet bij het sluiten van een zaak aanwezig zijn, dus je kan de zaak niet sluiten want je kan immers 'einddatum' niet in de root van het object-read antwoord in de Objecten API zetten.
2. De object URL hoeft niet gedefinieerd te zijn, in dat geval wordt er gekeken naar het datum_kenmerk als attribuut van local_object, wat het specifiekere zaakobject-data dan bevat. Echter hier speelt een vergelijkbaar probleem aangezien je bijvoorbeeld voor 'object_type': 'overige' in de 'overige_data' zou willen kijken. Dus dit zou alleen voor WozWaarde mogelijk zijn (maar daarvoor is het datumveld 9 ipv 10 characters?)? Voor object type 'Overige' moet je in 'overige_data' kijken, echter dat doet van wat ik heb gezien get_brondatum() niet. 

**Expected behavior**
1. Je zou of in het resultaattype de diepere laag moeten kunnen opgeven (bijvoorbeeld: 'record.einddatum' of 'record/einddatum') en get_brondatum() zou die diepere attributen moeten kunnen opvragen. 
2. Vergelijkbaar zou je bij de lokale objecten in 'Overige.overige_data' de einddatum willen controleren. Al dan niet in combinatie met diepere datum_kenmerk verwijzigingen zoals hierboven. 

Als alternatief zou je ook alle attributen in het JSON object kunnen aflopen, maar dan zou je meerdere datum_kenmerk attributen tegen kunnen komen dus dat wil je waarschijnlijk niet.

### PR DESCRIPTION
**Pull request by stevenbal**
Fixes #1353

**Changes**

* Allow nested datumkenmerk for afleidingswijze zaakobject

### CODE REVIEW

File: src/openzaak/components/catalogi/models/resultaattype.py [hunk 1]
```diff
@@ -145,7 +145,9 @@ class ResultaatType(ETagMixin, OptionalGeldigheidMixin, models.Model):
         blank=True,
         help_text=_(
             "Naam van de attribuutsoort van het procesobject dat "
-            "bepalend is voor het einde van de procestermijn."
+            "bepalend is voor het einde van de procestermijn. "
+            "In het geval van afleidingswijze zaakobject kunnen nested attributen "
```
Comment (annashamray) [inline src/openzaak/components/catalogi/models/resultaattype.py:149]: This behavior is different from the standard. Could you please make a note about it here, perhaps mark the last sentence as experimental?

File: src/openzaak/components/catalogi/models/resultaattype.py [hunk 2]
```diff
@@ -145,7 +145,9 @@ class ResultaatType(ETagMixin, OptionalGeldigheidMixin, models.Model):
...
+            "gespecificeerd worden door middel van `/` (bijv. `some/nested/path`)"
```
Comment (annashamray) [inline src/openzaak/components/catalogi/models/resultaattype.py:150]: But you expect `.` not `/` as a separator?

**Review by annashamray — CHANGES_REQUESTED** [review]
Could you also please update Catalogi OAS?

File: src/openzaak/components/catalogi/models/resultaattype.py (hunk 2 above)
Comment (stevenbal) [reply src/openzaak/components/catalogi/models/resultaattype.py:150]: I split by `/` to create a path, so nested attributes are specified with `/`: https://github.com/open-zaak/open-zaak/pull/1683/files#diff-4edd9bb7115bb5ddd0eb9986700a0fad6adf34a7d519b394b85359c220d292a5R146

I also added tests for `.` to check if we do still support attributes with dots in them:

```
{
    "some.attribute": "foo"
}
```

File: src/openzaak/components/catalogi/models/resultaattype.py (hunk 1 above)
Comment (stevenbal) [reply src/openzaak/components/catalogi/models/resultaattype.py]: added it to the helptext

**Review by annashamray — APPROVED** [review]

**Pull request merged**